src/pybel/schema/__init__.py

```diff
@@ -3,64 +3,29 @@
 """Validation for PyBEL data."""
 
 import json
+import logging
 import os
 from typing import Any, Mapping, Optional, Tuple
 
 import jsonschema
 
 __all__ = ['is_valid_node']
 
-HERE = os.path.abspath(os.path.dirname(__file__))
-
-
-def _load_schema(filename: str) -> Mapping[str, Any]:
-    """
-    Load a schema from its filename.
+logger = logging.getLogger(__name__)
 
-    :param filename: The relative path to the schema, e.g. "variant.schema.json"
-    """
-    path = os.path.join(HERE, filename)
-    with open(path) as json_schema:
-        schema = json.load(json_schema)
-    return schema
-
-
-def _validator(filename: str) -> jsonschema.Draft7Validator:
-    """
-    Return a validator that checks against a given schema.
+HERE = os.path.abspath(os.path.dirname(__file__))
+SCHEMA_PATH = os.path.join(HERE, "node.schema.json")
 
-    :param filename: The relative path to the schema, e.g. "variant.schema.json"
-    """
-    schema = _load_schema(filename)
-    # To use schemas from other files, jsonschema needs to know where the references point to, so
-    # create a resolver that directs any references (like "fusion.schema.json") to the schema's dir
-    schema_uri = 'file://' + HERE + '/'
-    resolver = jsonschema.RefResolver(base_uri=schema_uri, referrer=__file__)
-    return jsonschema.Draft7Validator(schema, resolver=resolver)
+# Load the top level schema
+with open(SCHEMA_PATH) as json_schema:
+    SCHEMA = json.load(json_schema)
 
-
-def _get_schema(node: Mapping[str, Any]) -> Tuple[str, bool]:
-    """
-    Get the filename for the appropriate schema for a given node.
-
-    :return: The filename for the schema and an error bool.
-    """
-    err = False
-    filename = ""
-    # Check if the node is a subclass of FusionBase
-    if 'fusion' in node.keys():
-        filename = 'fusion.schema.json'
-    # Check if the "function" is present (if not, the node is invalid)
-    elif 'function' in node.keys():
-        if node['function'] == 'Reaction':
-            filename = 'reaction.schema.json'
-        elif node['function'] in ['Complex', 'Composite']:
-            filename = 'list_abundance.schema.json'
-        else:
-            filename = 'base_abundance.schema.json'
-    else:
-        err = True
-    return filename, err
+# To use schemas from other files, jsonschema needs to know where the references point to, so
+# create a resolver that directs any references (like "fusion.schema.json") to the schema's dir
+schema_uri = 'file://' + HERE + '/'
```

Comment:
this makes me uneasy. Why don't you check out [`pathlib`](https://docs.python.org/3/library/pathlib.html) to find the safer version for building filepaths (I think [this function](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_uri) will do the trick)

Reply:
Fixed!